validator/slashing-protection/proposer_protect.go

```diff
@@ -0,0 +1,49 @@
+package slashingprotection
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+	ethpb "github.com/prysmaticlabs/ethereumapis/eth/v1alpha1"
+	"github.com/prysmaticlabs/prysm/beacon-chain/core/helpers"
+	"github.com/prysmaticlabs/prysm/shared/params"
+)
+
+// IsSlashableBlock checks if signed beacon block is slashable against
+// a validator's slashing protection history and against a remote slashing protector if enabled.
+func (s *Service) IsSlashableBlock(
+	ctx context.Context, block *ethpb.SignedBeaconBlock, pubKey [48]byte, domain *ethpb.DomainResponse,
+) (bool, error) {
+	if block == nil || block.Block == nil {
+		return false, errors.New("received nil block")
+	}
+	metricsKey := fmt.Sprintf("%#x", pubKey)
+	existingSigningRoot, err := s.validatorDB.ProposalHistoryForSlot(ctx, pubKey[:], block.Block.Slot)
```

Comment:
Move this to after the remote protector check. If remote protector is enabled, this data isn't used.